coordinator/internal/logic/provertask/prover_task.go

```diff
@@ -115,3 +118,22 @@ func (b *BaseProverTask) getHardForkNumberByName(forkName string) (uint64, error
 
 	return hardForkNumber, nil
 }
+
+var (
+	getTaskCounterInitOnce sync.Once
+	getTaskCounterVec      *prometheus.CounterVec = nil
+)
+
+func newGetTaskCounterVec(factory promauto.Factory, taskType string) *prometheus.CounterVec {
+	getTaskCounterInitOnce.Do(func() {
+		getTaskCounterVec = factory.NewCounterVec(prometheus.CounterOpts{
+			Name: "coordinator_get_task_count",
+			Help: "Multi dimensions get task counter.",
+		}, []string{"task_type",
+			coordinatorType.LabelProverName,
+			coordinatorType.LabelProverPublicKey,
+			coordinatorType.LabelProverVersion})
+	})
+
+	return getTaskCounterVec.MustCurryWith(prometheus.Labels{"task_type": taskType})
```

Comment:
Why use prune here?

Reply:
Since the "task_type" label just appears several lines above. The must here will not cause any problem.
The once here is to ensure that the metric(shared by batch and chunk) only be registered once.